rust/src/main.rs

```diff
@@ -76,37 +109,147 @@ fn setup_logging() -> Result<(), fern::InitError> {
         }).level(level_filter)
         .chain(io::stderr());
 
-    let path_result = path_for_log_file(XI_LOG_FILE);
-    // If the logging_file_path returned successfully, add the logfile capability to fern
-    if let Ok(logging_file_path) = &path_result {
+    if let Some(logging_file_path) = &logging_path_result {
+        // Try to create the parent directories for the logging file in the logging file path
+        create_log_directory(&logging_file_path)?;
+
+        // Attach it to fern
         fern_dispatch = fern_dispatch.chain(fern::log_file(logging_file_path)?);
-    }
+    };
 
     // Start fern
     fern_dispatch.apply()?;
+    info!("Logging with fern is setup");
 
-    // If the logging_file_path returned an error, print it with fern/logs
-    if let Err(e) = &path_result {
-        let message = "There was an issue getting the path for the log file";
-        warn!("{}: {:?}, falling back to stderr.", message, e);
+    // Log details of the logging_file_path result using fern/log
+    // Either logging the path fern is outputting to or the error from obtaining the path
+    match &logging_path_result {
+        Some(logging_file_path) => info!(
+            "Logging to the following file: {}",
+            logging_file_path.display()
+        ),
+        None => warn!("There was no path supplied for the log file, falling back to stderr."),
```

Comment:
"No path supplied for the log file, ..." or "No path was supplied for the log file, ..." sounds better IMHO.
"Log file without a name requested; only logging to stderr" might be more descriptive though.